src/fhirtypes/common.ts

```diff
@@ -855,14 +855,16 @@ export function replaceReferences<T extends AssignmentRule | CaretValueRule>(
       }
     }
   } else if (value instanceof FshCode) {
-    const [system, ...versionParts] = value.system?.split('|') ?? [];
-    const version = versionParts.join('|');
-    const codeSystem = tank.fish(system, Type.CodeSystem);
-    const codeSystemMeta = fisher.fishForMetadata(system, Type.CodeSystem);
+    const codeSystem = fishInTankBestVersion(tank, value.system, Type.CodeSystem);
+    const codeSystemMeta = fishForMetadataBestVersion(fisher, value.system, Type.CodeSystem);
     if (codeSystemMeta) {
       clone = cloneDeep(rule);
       const assignedCode = getRuleValue(clone) as FshCode;
-      assignedCode.system = `${codeSystemMeta.url}${version ? `|${version}` : ''}`;
+      const [, ...versionParts] = value.system?.split('|') ?? [];
+      const version = versionParts.join('|');
+      assignedCode.system = `${codeSystemMeta.url}${
+        !version || codeSystemMeta.url.endsWith(version) ? '' : `|${version}`
+      }`;
```

Comment:
Another opportunity for the regex replace (I think):
```ts
assignedCode.system = value.system.replace(/^[^|]+/, codeSystemMeta.url);
```

Reply:
I wonder if it would be convenient or annoying to wrap the regex replace in a helper function...

Reply:
Part of [70e31fc](https://github.com/FHIR/sushi/pull/1275/commits/70e31fc1b4bdb5eaab6119f21d08e44e038c24db).

And I'm leaning that it would be annoying to wrap in a helper function... so didn't do that.